lib/sherlock_homes/property_finder.rb

```diff
@@ -0,0 +1,25 @@
+module SherlockHomes
+
+  class ZillowServiceError < RuntimeError; end
+  class AddressResolveError < RuntimeError; end
+  class NoPropertyError < RuntimeError; end
+
+  class PropertyFinder
+
+    def initialize
+      Rubillow.configure { |c| c.zwsid = SherlockHomes.configuration.zillow_key }
+    end
+
+    def search(address: '', citystatezip: '')
+      property = Rubillow::PropertyDetails.deep_search_results(address: address, citystatezip: citystatezip)
```

Comment:
Rubillow is not invoking Zillow API if address or citystatezip is missing, and raises an ArgumentError. But we can do our own check if we need